uwblueprint/starter-code-v2
# Modify and clean up infra setup

## Notion ticket link

Closes #71 
Closes #109 



## Implementation description
* Set container names in docker-compose.yml so that they not system-dependent
* Move test database onto the same PostgreSQL server as dev database
* Add health check for DB service

Resources:
* https://stackoverflow.com/questions/46668233/multiple-databases-in-docker-and-docker-compose
* https://stackoverflow.com/a/55835081



## Steps to test
1. Copy updated secrets from #internal-tools-alerts in Slack into root .env file (will push to Vault after this PR is merged)
2. Remove previous volumes
```
docker-compose down --volumes
docker volume prune
```
3. Comment out `py-backend` in docker-compose.yml
4. Run the application, ensure that the backend successfully connects to the PostgreSQL database (server will crash if unsuccessful)
```
docker-compose up --build
```
5. Check that container names are as expected:
```
docker ps
```
6. Run backend tests to ensure successful connection to test database
```
docker exec -it scv2_ts_backend /bin/bash -c "yarn test"
```
7. Repeat steps 4-6 with the Python backend (comment out `ts-backend` in docker-compose.yml`). Test command for Python backend:
```
docker exec -it scv2_py_backend /bin/bash -c "pip install -e . && pytest"
```



## What should reviewers focus on?
* N/A


## Checklist
